Monitoring Guide - initial work

lzap opened the pull request:
This is replacement of my old PCP, there should not be any changes other than using our "upstreamize" script. Please review this first. Then I will followup with either commits or PRs as required. The goal is no changes at this point, then I want to update the document a bit and add new info about external PCP integration and new RHEL8 features.

melcorr commented on guides/doc-Monitoring_Guide/master.adoc:
```diff
@@ -0,0 +1,25 @@
+include::topics/proc_installing-pcp-packages.adoc[leveloffset=+1]
+include::topics/proc_configuring-pcp-data-collection.adoc[leveloffset=+2]
+include::topics/proc_enabling-access-to-metrics-via-web-ui.adoc[leveloffset=+2]
+include::topics/proc_verifying-pcp-configuration.adoc[leveloffset=+2]
```
remove this file - it adds nothing and will link to real docs from a proper introduction

lzap replied:
Okay.

melcorr commented on guides/doc-Monitoring_Guide/master.adoc:
```diff
@@ -0,0 +1,25 @@
+
+include::common/attributes.adoc[]
+include::topics/con_satellite_metrics_overview.adoc[leveloffset=+1]
+include::topics/con_performance-co-pilot.adoc[leveloffset=+1]
```
please remove this file - total garbage that adds nothing

lzap replied:
Only this component/file? It's introduction to PCP followed by introduction to PMDA.

melcorr commented on guides/doc-Monitoring_Guide/topics/proc_identifying-available-metrics.adoc:
```diff
@@ -0,0 +1,32 @@
+The output includes the file name of the active archive file, for example:
++
+----
+/var/log/pcp/pmlogger/satellite.example.com/20180814.00.10
```
does the upstremize script need to be updated to catch these, or is it best not to do that because when it is a directory, it may very well be either /foreman or /satellite without being updated downstream?

lzap replied:
I think we should handle these manually, this is special case I'd say where URL is in a path.

melcorr commented on guides/doc-Monitoring_Guide/topics/proc_retrieving-archived-metrics-using-cli.adoc:
```diff
@@ -0,0 +1,58 @@
+* To list disk writes for each partition, over the time period covered by the archive file:
++
+----
+# pmval --archive /var/log/pcp/pmlogger/satellite.example.com/20180816.00.10 \
```
here too

melcorr commented on guides/doc-Monitoring_Guide/topics/proc_retrieving-live-metrics-using-cli.adoc:
```diff
@@ -0,0 +1,31 @@
+.Example output
+----
+metric:    disk.partitions.write
+host:      satellite.example.com
```
Probably all that's happening here is it is missing the [attributes] markup

melcorr commented on guides/doc-Monitoring_Guide/topics/proc_verifying-pcp-configuration.adoc:
```diff
@@ -0,0 +1,25 @@
+
+.Example output from the `pcp` command:
+----
+Performance Co-Pilot configuration on satellite.example.com:
```
same here

lzap replied:
This file was deleted.

melcorr commented on guides/doc-Monitoring_Guide/topics/proc_verifying-pcp-configuration.adoc:
```diff
@@ -0,0 +1,25 @@
+----
+Performance Co-Pilot configuration on satellite.example.com:
+
+ platform: Linux satellite.example.com 3.10.0-862.3.3.el7.x86_64 #1 SMP Wed Jun 13 05:44:23 EDT 2018 x86_64
```
same here

lzap replied:
This file was deleted.

melcorr commented on guides/doc-Monitoring_Guide/master.adoc:
```diff
@@ -0,0 +1,25 @@
+include::topics/con_pcp-metrics.adoc[leveloffset=+1]
+include::topics/proc_identifying-available-metrics.adoc[leveloffset=+2]
+include::topics/con_retrieving-metrics.adoc[leveloffset=+1]
+include::topics/con_retrieving-metrics-using-cli.adoc[leveloffset=+2]
```
please remove this file

lzap replied:
Done.

melcorr commented on guides/doc-Monitoring_Guide/master.adoc:
```diff
@@ -0,0 +1,25 @@
+include::topics/proc_identifying-available-metrics.adoc[leveloffset=+2]
+include::topics/con_retrieving-metrics.adoc[leveloffset=+1]
+include::topics/con_retrieving-metrics-using-cli.adoc[leveloffset=+2]
+include::topics/proc_retrieving-live-metrics-using-cli.adoc[leveloffset=+3]
```
move this to [leveloffset=+2]

lzap replied:
Done.

lzap commented on guides/doc-Monitoring_Guide/topics/proc_configuring-pcp-data-collection.adoc:
```diff
@@ -0,0 +1,204 @@
+.. Install the Foreman Telemetry and `pcp-mmvstatsd` packages.
++
+----
+# satellite-maintain packages install foreman-telemetry pcp-mmvstatsd
```
Good catch.

lzap commented:
I've pushed the requested changes, now I am working on changes that are needed in order to support external monitoring.

lzap commented:
What I am gonna do now is shake things a bit, reorder chapters because PMs wants to support external monitoring. I will making things more simple and all contents I am going to write as minimum text as possible only sticking with technical things (commands). The rest can be added later.

lzap commented:
First version of my rewrite is up, it's not complete yet but I've simplified things a bit. The idea is that we now have two servers - Foreman server and monitoring server. Instructions and concepts differ on those two, therefore these are two big separate chapters.

Ignore grammar, just let me know if this goes in the good direction.

I have to say, many sentences do not make sense. Somebody took my original KBASE article which was probably bad grammar but it made sense and turned it into this. I've rewritten parts which were not correct.

melcorr commented:
This hasn't been updated in almost 2 years. Closing for now because I believe it will require a completely fresh take on Monitoring at this stage. Feel free to reopen if you disagree!
